build/Makefile

```diff
@@ -353,10 +357,20 @@ uninstall: $(ensure-build-image)
 
 
 # Build a static binary for the gameserver controller
+ifeq ($(WITH_ARM64), 1)
 build-controller-binary: $(ensure-build-image)
 	$(GO_BUILD_LINUX_AMD64) \
 	-tags $(GO_BUILD_TAGS) -o $(go_build_base_path)/cmd/controller/bin/controller \
 	$(go_rebuild_flags) $(go_version_flags) -installsuffix cgo $(agones_package)/cmd/controller
+	$(GO_BUILD_LINUX_ARM64) \
```

Comment:
Two fun things here:

1. The second build will overwrite the first, since the binary name for both builds is `controller`. We should probably replicate the pattern we have with the sdk and include some version of the arch in the file name.
1. I expect this is why the CI is failing, because we are sending an ARM64 binary to run in a AMD64 docker image.
1. If we change the filename, we'll need to also update the Docker images. I think for this PR we can hardcode that for now, and then come back around and fix it up when we do multi-arch images.
1. It would be good to copy the pattern of a seperate target for each arch type, like we do with the `sdk` targets, and then have those targets as dependencies of this one - that way they can be built in parrallel 😄

Reply:
I really don't like harcoded variable.
WDYT about --build_arg option ?

Reply:
Will do harcoded var for now, and when building arm64 img, will use this pattern